lib/services/appmanager-service.ts

```diff
@@ -138,7 +138,10 @@ class AppManagerService implements IAppManagerService {
 
 			this.$project.importProject().wait();
 			this.$logger.printInfoMessageOnSameLine("Publishing patch for " + platforms.join(", ") + "...");
-			this.$progressIndicator.showProgressIndicator(this.$server.tam.uploadPatch(this.$project.projectData.ProjectName, this.$project.projectData.ProjectName, <any>{ Platforms: platforms }), 2000).wait();
+
+			let patchData = <any> { Platforms: platforms, IsMandatory: this.$options.isMandatory };
```

Comment:
The any cast was present in the original implementation -- my understanding is that it was used in order to make the { Platforms: platforms } object assignable to Server.PatchData. As Server.PatchData.Platforms is of type Server.DevicePlatform[] while the platforms argument in this method is of type string[].

I personally could not find a better and not too verbose way of converting string[] to Server.DevicePlatform[] so I kept the any cast. One option is to change Server.PatchData.Platforms type to string[], then there will be no need for any cast (either way some of the api now use the platform concept as string, and other parts -- as Server.DevicePlatform const enum.